apps/protocol-frontend/src/components/DaoDashboardShell.tsx

```diff
@@ -1,24 +1,110 @@
   daoId: number;
 }
-
-const startDate = new Date('7/1/2022'); // temporary hardcoded value while working on the datepicker component
-const endDate = new Date('8/1/2022'); // temporary hardcoded value while working on the datepicker component
+const TODAY_DATE = new Date();
 
 const DaoDashboardShell = ({ daoName, daoId }: DaoDashboardShellProps) => {
+  const [showCustomDatePicker, setShowCustomDatePicker] = useState(false);
+
+  const [startDate, setStartDate] = useState<Date>(new Date(TODAY_DATE));
+  const [endDate, setEndDate] = useState<Date>(
+    new Date(subWeeks(TODAY_DATE, 4)),
+  );
+  const [dateRange, setDateRange] = useState<{
+    label: string;
+    value: number;
+  } | null>(null);
+
+  const dateRangeOptions = [
+    {
+      value: 'Custom',
+      label: 'Custom',
+    },
+    { value: 1, label: 'Last Week' },
+    { value: 4, label: 'Last Month' },
+    { value: 12, label: 'Last Quarter' },
+    { value: 52, label: 'Last Year' },
+  ];
+
+  const dateChangeHandler = (selectedDateOffset: number | string) => {
+    if (selectedDateOffset === 'Custom') {
+      setShowCustomDatePicker(true);
+    }
+    setEndDate(TODAY_DATE);
+    setStartDate(
+      subWeeks(
+        TODAY_DATE,
+        typeof selectedDateOffset === 'number' ? selectedDateOffset : 0,
+      ),
+    );
+  };
+
   return (
     <Box
       paddingY={{ base: '4', md: '8' }}
       paddingX={{ base: '4', md: '8' }}
       color="gray.700"
       width="100%"
     >
-      <PageHeading>{daoName}</PageHeading>
+      <Flex
+        direction={{ base: 'column', lg: 'row' }}
+        justifyContent="space-between"
+        alignItems="center"
+      >
+        <PageHeading>{daoName}</PageHeading>
+        <Flex
+          flexBasis={{ base: '100%', lg: '60%' }}
+          direction={{ base: 'column', lg: 'row' }}
+          alignItems="center"
+          justifyContent={{ base: 'flex-start', lg: 'flex-end' }}
+          gap={2}
+        >
+          <Flex
+            direction={{ base: 'column', lg: 'row' }}
+            alignItems="center"
+            justifyContent={{ base: 'flex-start', lg: 'flex-end' }}
+            width="100%"
+            flexBasis={{ base: '100%', lg: '60%' }}
+            flexGrow="1"
+            gap={{ base: 0, lg: 2 }}
+          >
+            <Box visibility={showCustomDatePicker ? 'inherit' : 'hidden'}>
+              <ControlledDatePicker
+                selected={startDate}
+                onChange={dates => {
+                  const datesArray = dates as Date[];
+                  const [start, end] = datesArray;
+                  setStartDate(start);
+                  setEndDate(end);
+                }}
+                startDate={startDate}
+                endDate={endDate}
+                maxDate={new Date(TODAY_DATE)}
+              />
+            </Box>
+            <ControlledSelect
+              isSearchable={false}
+              defaultValue={dateRangeOptions.find(date => date.value === 4)}
```

Comment:
Can we use a const for 4? something like Month I guess? It could make the code more readable